src/PhpParser/NodeFinder/PropertyFetchFinder.php

```diff
@@ -34,34 +50,53 @@ public function findPrivatePropertyFetches(Property | Param $propertyOrPromotedP
             return [];
         }
 
-        $classLikesToSearch = $this->nodeRepository->findUsedTraitsInClass($classLike);
-        $classLikesToSearch[] = $classLike;
-
         $propertyName = $this->resolvePropertyName($propertyOrPromotedParam);
         if ($propertyName === null) {
             return [];
         }
 
-        /** @var PropertyFetch[]|StaticPropertyFetch[] $propertyFetches */
-        $propertyFetches = $this->betterNodeFinder->find($classLikesToSearch, function (Node $node) use (
-            $propertyName,
-            $classLikesToSearch
-        ): bool {
-            // property + static fetch
-            if (! $node instanceof PropertyFetch && ! $node instanceof StaticPropertyFetch) {
-                return false;
+        $className = (string) $this->nodeNameResolver->getName($classLike);
+        if (! $this->reflectionProvider->hasClass($className)) {
+            /** @var PropertyFetch[]|StaticPropertyFetch[] $propertyFetches */
+            $propertyFetches = $this->findPropertyFetchesInClassLike($classLike->stmts, $propertyName);
+            return $propertyFetches;
+        }
+
+        $classReflection = $this->reflectionProvider->getClass($className);
+
+        $classLikes = $classReflection->getTraits(true);
+        $nodes = [$classLike];
+        foreach ($classLikes as $classLike) {
+            $fileName = $classLike->getFileName();
+            if (! $fileName) {
+                continue;
             }
 
-            // is it the name match?
-            if (! $this->nodeNameResolver->isName($node, $propertyName)) {
-                return false;
+            $fileContent = $this->smartFileSystem->readFile($fileName);
+            /** @var Stmt[] $parsedNodes */
+            $parsedNodes = $this->parser->parse($fileContent);
+
+            $smartFileInfo = new SmartFileInfo($fileName);
+            $file = new File($smartFileInfo, $smartFileInfo->getContents());
+
+            /** @var Stmt[] $allNodes */
+            $allNodes = $this->nodeScopeAndMetadataDecorator->decorateNodesFromFile($file, $parsedNodes);
+            $traitName = $classLike->getName();
+
+            /** @var Trait_|null $trait */
+            $trait = $this->betterNodeFinder->findFirst(
+                $allNodes,
+                fn (Node $node): bool => $node instanceof Trait_ && $this->nodeNameResolver->isName($node, $traitName)
+            );
+
+            if (! $trait instanceof Trait_) {
+                continue;
             }
 
-            $currentClassLike = $node->getAttribute(AttributeKey::CLASS_NODE);
-            return in_array($currentClassLike, $classLikesToSearch, true);
-        });
+            $nodes[] = $trait;
+        }
```

Comment:
Could you extract this logic to AstResolver? Something like `parseClasdReflectionTraits()`?

Reply:
implemented https://github.com/rectorphp/rector-src/pull/481/commits/cec33c5d2730c63bb738ef8fa8a8e8d302f59196